verticapy/learn/memmodel.py

```diff
@@ -0,0 +1,1036 @@
+    -------
+    numpy.array
+        Transformed data
+    """
+    check_types([("X", X, [list],), 
+                 ("values", values, [list],),
+                 ("method", method, ["zscore", "robust_zscore", "minmax"],),])
+    a, b = np.array([elem[0] for elem in values]), np.array([elem[1] for elem in values])
+    if method == "minmax":
+        b = b - a
+    return (np.array(X) - a) / b
+
+# ---#
+def sql_from_normalizer(X: list, 
+                        values: list,
+                        method: str = "zscore",):
+    """
+    ---------------------------------------------------------------------------
+    Returns the SQL code needed to deploy a normalizer model using its attributes.
+
+    Parameters
+    ----------
+    X: list
+        Names or values of the input predictors.
+    values: list
+        List of tuples, including the model's attributes. These required tuple  
+        depends on the specified method:
+            'zscore': (mean, std)
+            'robust_zscore': (median, mad)
+            'minmax': (min, max)
+    method: str, optional
+        The model's category, one of the following: 'zscore', 'robust_zscore', or 'minmax'.
+
+    Returns
+    -------
+    list
+        SQL code
+    """
+    check_types([("X", X, [list],), 
+                 ("values", values, [list],),
+                 ("method", method, ["zscore", "robust_zscore", "minmax"],),])
+    assert len(X) == len(values), ParameterError("The length of parameter 'X' must be equal to the length of the list 'values'.")
+    sql = []
+    for i in range(len(X)):
+        sql += ["({} - {}) / {}".format(X[i], values[i][0], values[i][1] - values[i][0] if method == "minmax" else values[i][1],)]
+    return sql
+
+# ---#
+def transform_from_one_hot_encoder(X: list, 
+                                   categories: list,
+                                   drop_first: bool = False,):
+    """
+    ---------------------------------------------------------------------------
+    Transforms the data with a one-hot encoder model using the input attributes.
+
+    Parameters
+    ----------
+    X: list / numpy.array
+        Data to transform.
+    categories: list
+        List of the categories of the different input columns.
+    drop_first: bool, optional
+        If set to False, the first dummy of each category will be dropped.
+
+    Returns
+    -------
+    list
+        SQL code
+    """
+    check_types([("X", X, [list],), 
+                 ("categories", categories, [list],),
+                 ("drop_first", drop_first, [bool],),])
+    def ooe_row(X):
+        result = []
+        for idx, elem in enumerate(X):
+            for idx2, item in enumerate(categories[idx]):
+                if idx2 != 0 or not(drop_first):
+                    if str(elem) == str(item):
+                        result += [1]
+                    else:
+                        result += [0]
+        return result
+    return np.apply_along_axis(ooe_row, 1, X)
```

Comment:
What is the benefit of `apply_along_axis` here? Can't you directly run the code inside `transform_from_one_hot_encoder`?

Reply:
apply_along_axis will parallelize the computation for each row. The function above is just to transform one row. The global will work on the entire dataset.